src/compiler/checker.ts

```diff
@@ -877,7 +877,11 @@ namespace ts {
                 (source.flags | target.flags) & SymbolFlags.Assignment) {
                 Debug.assert(source !== target);
                 if (!(target.flags & SymbolFlags.Transient)) {
-                    target = cloneSymbol(resolveSymbol(target));
+                    const resolvedTarget = resolveSymbol(target);

```

Comment:
I'm not sure sure we should be merging unknownSymbol at all; I'll try to track down its origin to see whether there's some other symbol that should appear here.